src/vit-servicing-station/vit-servicing-station-lib/src/db/models/challenges.rs

```diff
@@ -132,13 +140,13 @@ pub mod test {
         populate_db_with_challenge_conn(challenge, &connection);
     }
 
-    pub fn populate_db_with_challenge_conn(
-        challenge: &Challenge,
-        connection: &PooledConnection<ConnectionManager<DbConnection>>,
-    ) {
-        diesel::insert_into(challenges::table)
-            .values(challenge.clone().values())
-            .execute(connection)
-            .unwrap();
+    pub fn populate_db_with_challenge_conn(challenge: &Challenge, connection: &DbConnection) {
+        q!(
```

Comment:
is it possible to test the q macro or is it infallible?

Reply:
I don't know if I got what you mean but It shouldn't fail at generating the code. Errors like that would be caught at compile time. What can be done, I believe, is test if it generates the expected code.

Reply:
ok cool, sorry yes it should be caught at compile time. no worries.